realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -171,6 +165,45 @@ void BaseRealSenseNode::publishTopics()
     ROS_INFO_STREAM("RealSense Node Is Up!");
 }
 
+void BaseRealSenseNode::initializeFormatsMaps()
+{
+    // from rs2_format to OpenCV format
+    // https://docs.opencv.org/3.4/d1/d1b/group__core__hal__interface.html
+    // https://docs.opencv.org/2.4/modules/core/doc/basic_structures.html
+    // CV_<bit-depth>{U|S|F}C(<number_of_channels>)
+    // where U is unsigned integer type, S is signed integer type, and F is float type.
+    // For example, CV_8UC1 means a 8-bit single-channel array,
+    // CV_32FC2 means a 2-channel (complex) floating-point array, and so on.
+    rs_format_to_cv_format[RS2_FORMAT_Y8] = CV_8UC1;
+    rs_format_to_cv_format[RS2_FORMAT_Y16] = CV_16UC1;
+    rs_format_to_cv_format[RS2_FORMAT_Z16] = CV_16UC1;
+    rs_format_to_cv_format[RS2_FORMAT_RGB8] = CV_8UC3;
+    rs_format_to_cv_format[RS2_FORMAT_BGR8] = CV_8UC3;
+    rs_format_to_cv_format[RS2_FORMAT_RGBA8] = CV_8UC4;
+    rs_format_to_cv_format[RS2_FORMAT_BGRA8] = CV_8UC4;
+    rs_format_to_cv_format[RS2_FORMAT_YUYV] = CV_8UC2;
```

Comment:
What about UYVY?

Reply:
Added

Reply:
I also need to add format of Y10BPack, will test this tomorrow with D415.

Reply:
After investigating, Y10BPack is not supported anymore.